sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -322,54 +333,21 @@ namespace Azure { namespace Core { namespace Http {
 
   public:
     /**
-     * @brief Construct a Bearer Token authentication policy with single authentication scope.
-     *
-     * @param credential A #TokenCredential to use with this policy.
-     * @param scope Authentication scope.
-     */
-    explicit BearerTokenAuthenticationPolicy(
-        std::shared_ptr<TokenCredential const> credential,
-        std::string scope)
-        : m_credential(std::move(credential))
-    {
-      m_scopes.emplace_back(std::move(scope));
-    }
-
-    /**
-     * @brief Construct a Bearer Token authentication policy with multiple authentication scopes.
-     *
-     * @param credential A #TokenCredential to use with this policy.
-     * @param scopes A vector of authentication scopes.
-     */
-    explicit BearerTokenAuthenticationPolicy(
-        std::shared_ptr<TokenCredential const> credential,
-        std::vector<std::string> scopes)
-        : m_credential(std::move(credential)), m_scopes(std::move(scopes))
-    {
-    }
-
-    /**
-     * @brief Construct a Bearer Token authentication policy with multiple authentication scopes.
-     *
-     * @tparam A type of scopes sequence iterator.
+     * @brief Construct a Bearer Token authentication policy.
      *
      * @param credential A #TokenCredential to use with this policy.
-     * @param scopesBegin An iterator pointing to begin of the sequence of scopes to use.
-     * @param scopesEnd An iterator pointing to an element after the last element in sequence of
-     * scopes to use.
+     * @param getTokenOptions #GetTokenOptions.
      */
-    template <typename ScopesIterator>
     explicit BearerTokenAuthenticationPolicy(
         std::shared_ptr<TokenCredential const> credential,
-        ScopesIterator const& scopesBegin,
-        ScopesIterator const& scopesEnd)
-        : m_credential(std::move(credential)), m_scopes(scopesBegin, scopesEnd)
+        GetTokenOptions getTokenOptions)
```

Comment:
Wonder if getTokenOptions should be optional and have a default init just like we did on other policies like the transport and telemetry...

```suggestion
        GetTokenOptions getTokenOptions = GetTokenOptions ())
```

Reply:
+1, The default for options parameters follows the C++ design guidelines

Reply:
This was brought up earlier: https://github.com/Azure/azure-sdk-for-cpp/pull/1527#discussion_r567172776

Reply:
The reason I did not follow the pattern is that GetTokenOptions has scopes. A policy without scopes set is useless.

Reply:
Not fixing this, please reply if you insist, let's talk.

Reply:
Is it there is no valid set of default scopes we could set for scope?

Reply:
Scopes currently used do look like this: `https://storage.azure.com/.default`, `https://vault.azure.net/.default`.

Reply:
I.e. scope does include service name: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#scopes-and-permissions

Reply:
Also, from https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#the-default-scope: "The scope is constructed by using the resource URI and /.default." So, it is always some resource URI, I see no reasonable default.

Reply:
Thanks for the link.  That implies there are mandatory parameters to create the token options.  Not so keen on the name being options since they are required parameters....  
cc: @schaabs

Reply:
`TokenRequestParameters`? `TokenRequestContext`?
I am not fluent enough in English, does "options" that heavy imply things being non-mandatory, or is it more about following the pattern that is seen in other places across the SDK?